### DISCUSSION TITLE
Aug 9, 2022 Community Call

### DISCUSSION THREAD

**Discussion by expede** [ipvm-wg/spec#2] [Community Call]
Sign up: https://lu.ma/ipvm

# Suggested Agenda

_Let us know in the comments what you'd like to discuss! Questions about using UCAN, use cases in the wild, short demos, etc._

* Intros
* Recap Iceland for new folks
* Comms channels
  * Luma
  * Discord (Slack bridge coming)
  * GH Discussions (here!)
* Project goals
* Running code & rough consensus
* Review of Iceland kickoff

# Video

<a href="https://vimeo.com/738011441"><img width="979" alt="Screen Shot 2022-08-09 at 14 15 44" src="https://user-images.githubusercontent.com/1052016/183763183-ab07e418-a8fc-4590-8ad2-0eb8b96e23f9.png"></a>

# Chat Log

```
00:13:54	Brooklyn Zelenka (@expede):	https://github.com/ipvm-wg/spec/discussions/2
00:37:07	Adin Schmahmann:	https://github.com/ipld/ipld/pull/226/files
00:49:57	Oleg Silkin:	Is there a link to the autocodec proposal?
00:50:24	Brooklyn Zelenka (@expede):	Not formal, but here's some slides https://noti.st/expede/CWHgTE/ipld-autocodec
00:50:36	Oleg Silkin:	Thanks Brooklyn!
00:51:57	Brooklyn Zelenka (@expede):	Also Oleg — here's the high level IPVM proposal that I gave, but is not "the" canonical IPVM https://noti.st/expede/oq0ULd/ipvm-interplanetary-vm
00:52:41	Oleg Silkin:	Thanks Brooklyn! I read through it last week, it's very cool!
00:52:47	Brooklyn Zelenka (@expede):	Thanks!
00:59:57	Oleg Silkin:	I have to drop off, thanks all!
01:00:01	Brooklyn Zelenka (@expede):	Thanks!
01:00:40	Adin Schmahmann:	https://github.com/ipfs/specs/pull/305
01:13:07	James Walker (@walkah):	https://github.com/ipvm-wg/spec/discussions/2
```

# Notes

**Comment by RangerMauve** [comment]
Excited to see the recording! Sadly couldn't make it to the call due to other obligations. 😁

**Comment by softwareplumber** [comment]
Attendees:

- Brooklyn Zelenka - CTO at Fission, Working Group Lead 
- Adin Schmahmann - Kubo maintainer 
- Jonathan Essex - Engineer at Fission
- Oleg Silkin - Associate Software Engineer at RedHat. Decentralized storage.
- Hannah Howard - Filecoin/Go-IPFS 
- Melanie Riise - FVM Team. Community working group Web Assembly
- James Walker - Engineering Management at Fission

[Introductions]

Process: [https://github.com/ipvm-wg](https://github.com/ipvm-wg) is the place for long-term discussion

Brooke: (Interests/Contraints/Carry-Over from Rekjavik):

- Object Capabilities (OCAP)
- Agoric [ [https://github.com/Agoric](https://github.com/Agoric) ]
- Spritely [ [https://spritelyproject.org/](https://spritelyproject.org/) ]
- Distributed Message Bus / Differential Data Flow - intersection with Fission work
- Function Addressed Network Calls - PLN prior art [ Can’t find reference ]

Adin: 
Lots of things we might want from WASM in IPLD. Universal CODECs and ADLs. Issues with WASM ABIs for other language.
There’s more needed on the IPLD side - way of signaling ADLS.

Hannah:
How is IPLD represented in WASM?

Adin:
current state for codec: https://github.com/ipld/ipld/pull/226/files
Won’t work for ADLs - too big. Need partial representation of structure.

Hannah:
issues such as integer sizes likely to be influenced by WASM

[ Some generalized hating on floats - issues due to nondeterministic/platform dependent behaviors ]

[ Discussion focused on ADL use case ]

Melanie:
Suggest re-use C apis

Jonathan: 
Re-use prior art from XML processing stacks

Adin: 
IPFS methods have rich implicit behaviors triggered by the signal - ‘IPFS://’.

[ Discussion surrounding autocodecs:  [https://noti.st/expede/CWHgTE/ipld-autocodec](https://noti.st/expede/CWHgTE/ipld-autocodec) ]

[ Brooke would like feedback! ]

[ Discussion of signaling issues related to autocodes/ADLs ]

Jonathan:
Declarative versus Imperative signals, expressing intent of block producer vs requirement of block consumer

Melanie:
How to signal - undeveloped part of IPLD

Jonathan:
Single type signal at root of graph implies static typing; absent signal implies duck typing; to mix the two we need in-band signaling

Adin: 
Look at the Version 2 CID spec: https://github.com/ipfs/specs/pull/305

Hannah: 
what’s this for, how do we get to something deliverable

Brooke:
get to working code ASAP, but there are many use cases. Topics that came out of Rekjavik:

- IPLD Calling Convention
- WASM specific IDL
- Execution bounding
- IPLI loading
- Hypervisor
- IPLD-Based Linking

Hannah: 
What do you mean by Calling Via IPLD

Brooke: 
Composing WASM modules + Invocation Spec

Adin: 
Useful context: [https://github.com/ipld/ipld/tree/master/notebook/exploration-reports](https://github.com/ipld/ipld/tree/master/notebook/exploration-reports)

**Comment by BigLep** [comment]
@expede  or @softwareplumber : Awesome to see this happening.  For giving community visibility to it, could meeting invites be sent to ipfs.io_eal36ugu5e75s207gfjcu0ae84@group.calendar.google.com ?  This will cause it to appear on the IPFS Community Calendar.

Context: https://github.com/ipfs/ipfs-docs/issues/1237

**Reply by BigLep** [reply]
There is discussion about a different want to give visibility to the IPFS community rather than a google calendar happening here: https://discuss.ipfs.tech/t/community-calendar-options-other-than-google-cal/14954 .  In the meantime, I have added these events to the existing IPFS Community google calendar by personally joining the Luma events and inviting the IPFS community calendar.  No further action needed at the moment.
